operator/cmd/mesh/operator-remove.go

```diff
@@ -78,16 +78,26 @@ func operatorRemove(cmd *cobra.Command, args *RootArgs, orArgs *operatorRemoveAr
 		l.LogAndFatal(err)
 	}
 
+	switch {
 	// If the user is performing purge but also specified a revision, an error message
 	// should be displayed and do nothing
-	if orArgs.purge && orArgs.revision != "" {
+	case orArgs.purge && orArgs.revision != "":
 		orArgs.revision = ""
-		l.LogAndFatal("At most one of the --revision or --purge flags could be set\n")
-	} else if orArgs.revision == "default" {
+		l.LogAndPrint("Purge remove will remove all Istio operator controller, ignoring the specified revision\n")
+	case orArgs.purge:
 		orArgs.revision = ""
+	case orArgs.revision == "":
+		l.LogAndFatal("At least one of the --revision or --purge flags must be set")
```

Comment:
IMO this should not be the right layer to do the args validation in a cobra command, I think it's better to add these kind of thing to the `Args` part, and after successfully validate the args, the correct args can be passed into `RunE` and perform the rest operations.